rules/order/src/Rector/ClassLike/OrderMethodsByVisibilityRector.php

```diff
@@ -100,63 +98,28 @@ public function getNodeTypes(): array
     {
-        if ($node instanceof Node\Stmt\Interface_) {
+        if ($node instanceof Interface_) {
             return null;
         }
 
-        $classMethods = [];
-        $classMethodsByName = [];
-        foreach ($node->stmts as $position => $classStmt) {
-            if (! $classStmt instanceof ClassMethod) {
-                continue;
-            }
-
-            /** @var string $classMethodName */
-            $classMethodName = $this->getName($classStmt);
-            $classMethodsByName[$position] = $classMethodName;
-
-            $classMethods[$classMethodName]['name'] = $classMethodName;
-            $classMethods[$classMethodName][self::VISIBILITY] = $this->stmtOrder->getOrderByVisibility($classStmt);
-            $classMethods[$classMethodName]['abstract'] = $classStmt->isAbstract();
-            $classMethods[$classMethodName]['final'] = $classStmt->isFinal();
-            $classMethods[$classMethodName]['static'] = $classStmt->isStatic();
-            $classMethods[$classMethodName][self::POSITION] = $position;
-        }
-
-        $sortedMethods = $this->getMethodsSortedByVisibility($classMethods);
-        $methodsInPreferredOrder = $this->getMethodsInPreferredOrder($sortedMethods);
+        $currentMethodsOrder = $this->stmtOrder->getStmtsOfTypeOrder($node, ClassMethod::class);
+        $methodsInDesiredOrder = $this->getMethodsInDesiredOrder($node);
 
-        $oldToNewKeys = $this->stmtOrder->createOldToNewKeys($methodsInPreferredOrder, $classMethodsByName);
+        $oldToNewKeys = $this->stmtOrder->createOldToNewKeys($methodsInDesiredOrder, $currentMethodsOrder);
 
         return $this->stmtOrder->reorderClassStmtsByOldToNewKeys($node, $oldToNewKeys);
     }
 
-    private function getMethodsSortedByVisibility(array $classMethods): array
+    private function getMethodsInDesiredOrder(ClassLike $classLike): array
     {
-        uasort(
-            $classMethods,
-            function (array $firstArray, array $secondArray): int {
-                return [
-                    $firstArray[self::VISIBILITY],
-                    $firstArray['static'],
-                    $secondArray['abstract'],
-                    $firstArray['final'],
-                    $firstArray[self::POSITION],
-                ] <=> [
-                    $secondArray[self::VISIBILITY],
-                    $secondArray['static'],
-                    $firstArray['abstract'],
-                    $secondArray['final'],
-                    $secondArray[self::POSITION],
-                ];
-            }
-        );
-
-        return array_keys($classMethods);
+        $classMethods = $this->stmtVisibilitySorter->sortMethods($classLike);
+        $classMethods = array_keys($classMethods);
+
+        return $this->applyPreferredPosition($classMethods);
     }
 
-    private function getMethodsInPreferredOrder(array $sortedMethods): array
+    private function applyPreferredPosition(array $classMethods): array
```

Comment:
`array` would deserve types

I'm going to re-enable this in PHPStan, so it's covered by CI

Reply:
https://github.com/rectorphp/rector/pull/3936